src/Raven.Server/Documents/Indexes/Persistence/Lucene/LuceneIndexPersistence.cs

```diff
@@ -106,6 +109,29 @@ public LuceneIndexPersistence(Index index)
                 string fieldName = field.Key;
                 _suggestionsIndexSearcherHolders[fieldName] = new IndexSearcherHolder(state => new IndexSearcher(_suggestionsDirectories[fieldName], true, state), _index._indexStorage.DocumentDatabase);
             }
+
+            IndexSearcher CreateIndexSearcher(IState state)
+            {
+                lock (this)
+                {
+                    var reader = _lastReader;
+
+                    if (reader == null)
+                        reader = _lastReader = IndexReader.Open(_directory, readOnly: true, state);
+                    else
+                    {
+                        var newReader = reader.Reopen(state);
+                        if (newReader != reader)
+                            reader.DecRef(state);
+
+                        reader = _lastReader = newReader;
```

Comment:
You are exposing the `_lastReader` publicly before you called `IncRef`, a user calling `DecRef` can cause it to be closed, no?